WIP: Add the initial draft of the August Newsletter (#125)
fortran-lang/fortran-lang.org: _posts/2020-08-01-Fortran-Newsletter-August-2020.md

  
Ongoing work:

* [#117](https://github.com/fortran-lang/fortran-lang.org/issues/117): Adding a
  Benchmarks section, a new dedicated repository was created at
  https://github.com/fortran-lang/benchmarks and many details have been
  discussed in [issues](https://github.com/fortran-lang/benchmarks/issues) there

[Let us know](https://github.com/fortran-lang/fortran-lang.org/issues) if you have any suggestions for the website and its content.
We welcome any new contributors to the website and the tutorials page in particular - see the [contributor guide](https://github.com/fortran-lang/fortran-lang.org/blob/master/CONTRIBUTING.md) for how to get started.

## Fortran Standard Library

What's new in the Fortran Standard Library:
[#223](https://github.com/fortran-lang/stdlib/pull/223): the structure of the Fortran Standard Library has been modified for clarity and ease of use. With these changes, both experimental and stable procedures will reside together in the same modules. The status of the procedures (experimental vs stable) are documented in the code, in the specs, and in the [website](https://stdlib.fortran-lang.org/)

> 

Reviewer: ```suggestion
Main ongoing discussions:
* [#225](https://github.com/fortran-lang/stdlib/issues/225): Name convention for derived types in `stdlib`
* [#224](https://github.com/fortran-lang/stdlib/issues/224): Handling and propagating errors inside `stdlib`
* [#221](https://github.com/fortran-lang/stdlib/issues/221): API for a bitset data type
* [#201](https://github.com/fortran-lang/stdlib/issues/201): API for file system operations
```

  books about Fortran

* [#121](https://github.com/fortran-lang/fortran-lang.org/pull/121), [#122](https://github.com/fortran-lang/fortran-lang.org/pull/122), [#127](https://github.com/fortran-lang/fortran-lang.org/pull/127), [#128](https://github.com/fortran-lang/fortran-lang.org/pull/128): additional packages added to the Fortran-lang.org [packages](https://fortran-lang.org/packages) page
  
Ongoing work:

* [#117](https://github.com/fortran-lang/fortran-lang.org/issues/117): Adding a
  Benchmarks section, a new dedicated repository was created at
  https://github.com/fortran-lang/benchmarks and many details have been
  discussed in [issues](https://github.com/fortran-lang/benchmarks/issues) there

[Let us know](https://github.com/fortran-lang/fortran-lang.org/issues) if you have any suggestions for the website and its content.
We welcome any new contributors to the website and the tutorials page in particular - see the [contributor guide](https://github.com/fortran-lang/fortran-lang.org/blob/master/CONTRIBUTING.md) for how to get started.

## Fortran Standard Library

What's new in the Fortran Standard Library:

> [#223](https://github.com/fortran-lang/stdlib/pull/223): the structure of the Fortran Standard Library has been modified for clarity and ease of use. With these changes, both experimental and stable procedures will reside together in the same modules. The status of the procedures (experimental vs stable) are documented in the code, in the specs, and in the [website](https://stdlib.fortran-lang.org/)

Reviewer 2: ```suggestion
[#223](https://github.com/fortran-lang/stdlib/pull/223): the structure of the Fortran Standard Library has been modified for clarity and ease of use. With these changes, both experimental and stable procedures will reside together in the same modules. The status of the procedures (experimental vs stable) are documented in the code, in the specs, and in the [API docs](https://stdlib.fortran-lang.org/)
```

* [#224](https://github.com/fortran-lang/stdlib/issues/224): Handling and propagating errors inside `stdlib`
* [#221](https://github.com/fortran-lang/stdlib/issues/221): API for a bitset data type
* [#201](https://github.com/fortran-lang/stdlib/issues/201): API for file system operations

## Fortran Package Manager

What's new in fpm:

fpm is still in very early development, and we need as much help as we can get.
Here's how you can help today:

* Try to use it. Does it work? No? Let us know! Read the [fpm packaging guide](https://github.com/fortran-lang/fpm/blob/master/PACKAGING.md) to learn how to build your package with fpm.
* Browse the [open issues](https://github.com/fortran-lang/fpm/issues) and see if you can help implement any fixes or features.
* Adapt your Fortran package for fpm.
* Improve the documentation.

The short term goal of fpm is to make development and installation of Fortran packages with dependencies easier.
Its long term goal is to build a rich and decentralized ecosystem of Fortran packages and create a healthy
environment in which new open source Fortran projects are created and published with ease.

> 

Reviewer 2: ```suggestion

##  Fortran benchmarks

We created the [benchmarks repository](https://github.com/fortran-lang/benchmarks) with the goal to design and implement a comprehensive set of benchmarks.
The benchmarks will aim to compare the performance of various Fortran compilers, as well as the performance of canonical algorithms implemented in Fortran and different languages.
If you'd like to contribute in any way, be it the design, implementation, or testing of benchmarks, please join the ongoing discussion [here](https://github.com/fortran-lang/benchmarks/issues).

```

We welcome any new contributors to the website and the tutorials page in particular - see the [contributor guide](https://github.com/fortran-lang/fortran-lang.org/blob/master/CONTRIBUTING.md) for how to get started.

## Fortran Standard Library

What's new in the Fortran Standard Library:
[#223](https://github.com/fortran-lang/stdlib/pull/223): the structure of the Fortran Standard Library has been modified for clarity and ease of use. With these changes, both experimental and stable procedures will reside together in the same modules. The status of the procedures (experimental vs stable) are documented in the code, in the specs, and in the [API docs](https://stdlib.fortran-lang.org/)

Main ongoing discussions:
* [#225](https://github.com/fortran-lang/stdlib/issues/225): Name convention for derived types in `stdlib`
* [#224](https://github.com/fortran-lang/stdlib/issues/224): Handling and propagating errors inside `stdlib`
* [#221](https://github.com/fortran-lang/stdlib/issues/221): API for a bitset data type
* [#201](https://github.com/fortran-lang/stdlib/issues/201): API for file system operations

## Fortran Package Manager

What's new in fpm:

> * [fpm-registry](https://github.com/fortran-lang/fpm-registry) created, please see the README there how to contribute a package

Reviewer 2: ```suggestion
* We created the [fpm-registry](https://github.com/fortran-lang/fpm-registry) repository,
which serves as a registry of fpm-enabled Fortran packages.
Please see the README there to learn how to contribute a package.
For now, the registry is simply a list of fpm-enabled Fortran packages that you can use as a dependency in your `fpm.toml` file.
Soon, this registry will be used to generate detailed metadata that will be used by fpm to allow you to search for packages from the command-line, e.g. `fpm search <package>` or similar.
```

Main ongoing discussions:
* [#225](https://github.com/fortran-lang/stdlib/issues/225): Name convention for derived types in `stdlib`
* [#224](https://github.com/fortran-lang/stdlib/issues/224): Handling and propagating errors inside `stdlib`
* [#221](https://github.com/fortran-lang/stdlib/issues/221): API for a bitset data type
* [#201](https://github.com/fortran-lang/stdlib/issues/201): API for file system operations

## Fortran Package Manager

What's new in fpm:
* We created the [fpm-registry](https://github.com/fortran-lang/fpm-registry) repository,
which serves as a registry of fpm-enabled Fortran packages.
Please see the README there to learn how to contribute a package.
For now, the registry is simply a list of fpm-enabled Fortran packages that you can use as a dependency in your `fpm.toml` file.
Soon, this registry will be used to generate detailed metadata that will be used by fpm to allow you to search for packages from the command-line, e.g. `fpm search <package>` or similar.

> * [#146](https://github.com/fortran-lang/fpm/issues/146): we had a video discussion about the design of the new Fortran rewrite

Reviewer 2: ```suggestion
* [#146](https://github.com/fortran-lang/fpm/issues/146): We discussed the design of the new Fortran implementation of fpm in a video call. We agreed on the need for an intermediate package model which will allow for clean separation of fpm frontends (user interface, parsing, and semantics) and fpm backends (fpm itself, CMake, Make, etc.).
```

> 

Reviewer 2: ```suggestion
* [#131](https://github.com/fortran-lang/fpm/pull/131),
[#132](https://github.com/fortran-lang/fpm/pull/132),
[#139](https://github.com/fortran-lang/fpm/pull/139),
[#140](https://github.com/fortran-lang/fpm/pull/140),
[#142](https://github.com/fortran-lang/fpm/pull/142),
[#145](https://github.com/fortran-lang/fpm/pull/145),
[#147](https://github.com/fortran-lang/fpm/pull/147),
[#148](https://github.com/fortran-lang/fpm/pull/148),
[#151](https://github.com/fortran-lang/fpm/pull/151):
We merged several pull requests toward the Fortran fpm implementation. The Haskell implementation has moved to the `fpm/bootstrap` directory, and the Fortran implementation is developed in `fpm/fpm`. The Fortran fpm is, of course, an fpm package itself so it can be built by the Haskell fpm. Soon, it will be able to be build by itself. 

```

---
layout: post
title: "Fortran newsletter: August 2020"
category: newsletter
author: Ondřej Čertík, Milan Curcic, Laurence Kedward, Jérémie Vandenplas, Arjen Markus and Gary Klimowicz
---

Welcome to the August 2020 edition of the monthly Fortran newsletter.
The newsletter comes out on the first calendar day of every month
and details Fortran news from the previous month.

<ul id="page-nav"></ul>

## fortran-lang.org

> Work has continued on the Fortran-lang website, including:

Reviewer 2: Use active voice:

```suggestion
We continued the work on the Fortran-lang website, including:
```

@certik @LKedward please approve

  books about Fortran

* [#121](https://github.com/fortran-lang/fortran-lang.org/pull/121), [#122](https://github.com/fortran-lang/fortran-lang.org/pull/122), [#127](https://github.com/fortran-lang/fortran-lang.org/pull/127), [#128](https://github.com/fortran-lang/fortran-lang.org/pull/128): additional packages added to the Fortran-lang.org [packages](https://fortran-lang.org/packages) page
  
Ongoing work:

* [#117](https://github.com/fortran-lang/fortran-lang.org/issues/117): Adding a
  Benchmarks section, a new dedicated repository was created at
  https://github.com/fortran-lang/benchmarks and many details have been
  discussed in [issues](https://github.com/fortran-lang/benchmarks/issues) there

[Let us know](https://github.com/fortran-lang/fortran-lang.org/issues) if you have any suggestions for the website and its content.
We welcome any new contributors to the website and the tutorials page in particular - see the [contributor guide](https://github.com/fortran-lang/fortran-lang.org/blob/master/CONTRIBUTING.md) for how to get started.

## Fortran Standard Library

What's new in the Fortran Standard Library:

> [#223](https://github.com/fortran-lang/stdlib/pull/223): the structure of the Fortran Standard Library has been modified for clarity and ease of use. With these changes, both experimental and stable procedures will reside together in the same modules. The status of the procedures (experimental vs stable) are documented in the code, in the specs, and in the [API docs](https://stdlib.fortran-lang.org/)

Reviewer 2: ```suggestion
* [#223](https://github.com/fortran-lang/stdlib/pull/223): the structure of the Fortran Standard Library has been modified for clarity and ease of use. With these changes, both experimental and stable procedures will reside together in the same modules. The status of the procedures (experimental vs stable) are documented in the code, in the specs, and in the [API docs](https://stdlib.fortran-lang.org/)
```

  * [#163](https://gitlab.com/lfortran/lfortran/-/issues/163): Implement basic Fortran to C++ translation backend
  * [!410](https://gitlab.com/lfortran/lfortran/-/merge_requests/410): Make simple calculations work via LLVM in interactive mode
  * [!402](https://gitlab.com/lfortran/lfortran/-/merge_requests/402): Build ASR (Abstract Semantic Representation) properly for subroutines and functions

The short term goal is to get the C++ based production version of LFortran
matching most of the features from the Python prototype version and make a
public release. The long term goal is to build a modern Fortran compiler that
works with any production code and allows it to run efficiently on modern
hardware (CPUs and GPUs), both interactively and compiling to binaries, and
provide the basis for other tools such as the Fortran to C++ translation, editor
support, automatic documentation generation (and doctesting like in Python),
automatic formatting and others.

## Events

* [FortranCon 2020](https://tcevents.chem.uzh.ch/event/12) was held July 2 - 4.
with many interesting talks.
See the talk schedule
[here](https://tcevents.chem.uzh.ch/event/12/timetable/#20200702.detailed).

> FIXME: is recordings available?

Reviewer 2: ```suggestion
All presentations have been recorded and will be soon made available online by the FortranCon organizers.
```

The newsletter comes out on the first calendar day of every month
and details Fortran news from the previous month.

<ul id="page-nav"></ul>

## fortran-lang.org

Work has continued on the Fortran-lang website, including a new community page and additional tutorial content:

* [#120](https://github.com/fortran-lang/fortran-lang.org/pull/120): updated the
  [Book section](https://fortran-lang.org/learn/) with a comprehensive list of
  books about Fortran

Ongoing work:

* [#117](https://github.com/fortran-lang/fortran-lang.org/issues/117): Adding a
  Benchmarks section, a new dedicated repository was created at
  https://github.com/fortran-lang/benchmarks and many details have been
  discussed in [issues](https://github.com/fortran-lang/benchmarks/issues) there

[Let us know](https://github.com/fortran-lang/fortran-lang.org/issues) if you have any suggestions for the website and its content.
We welcome any new contributors to the website and the tutorials page in particular - see the [contributor guide](https://github.com/fortran-lang/fortran-lang.org/blob/master/CONTRIBUTING.md) for how to get started.

## Fortran Standard Library

What's new in the Fortran Standard Library:

> 

Reviewer: ```suggestion
[#223](https://github.com/fortran-lang/stdlib/pull/223): the structure of the Fortran Standard Library has been modified for clarity and ease of use. With these changes, both experimental and stable procedures will reside together in the same modules. The status of the procedures (experimental vs stable) are documented in the code, in the specs, and in the [website](https://stdlib.fortran-lang.org/)
```

> Some miscellaneous fixes and improvements:

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
Main ongoing discussions:
* [#225](https://github.com/fortran-lang/stdlib/issues/225): Name convention for derived types in`1stdlib`
* [#224](https://github.com/fortran-lang/stdlib/issues/224): Handling and propagating errors inside `stdlib`
* [#221](https://github.com/fortran-lang/stdlib/issues/221): APY for a bitset data type
* [#201](https://github.com/fortran-lang/stdlib/issues/201): API for file system operations
```

Author: in`1stdlib` -> in `stdlib`

Reviewer: > in`1stdlib` -> in `stdlib`

Changed!

Reviewer: Are these info suited for the newsletter? It not, it can be resolved.

Author: Also put space between in`stdlib`. 

It doesn't allow me to commit it for some reason --- but simply push the change in if you can.

## Fortran Package Manager

What's new in fpm:

fpm is still in very early development, and we need as much help as we can get.
Here's how you can help today:

* Try to use it. Does it work? No? Let us know! Read the [fpm packaging guide](https://github.com/fortran-lang/fpm/blob/master/PACKAGING.md) to learn how to build your package with fpm.
* Browse the [open issues](https://github.com/fortran-lang/fpm/issues) and see if you can help implement any fixes or features.
* Adapt your Fortran package for fpm.
* Improve the documentation.

The short term goal of fpm is to make development and installation of Fortran packages with dependencies easier.
Its long term goal is to build a rich and decentralized ecosystem of Fortran packages and create a healthy
environment in which new open source Fortran projects are created and published with ease.

## Events

* [FortranCon 2020](https://tcevents.chem.uzh.ch/event/12) was held July 2 - 4.
with many interesting talks.
See the talk schedule
[here](https://tcevents.chem.uzh.ch/event/12/timetable/#20200702.detailed).

> FIXME: is recordings available?

Reviewer 3: They said they would be made available on [their SWITCHtube](https://tube.switch.ch/organizations/5), but nothing there yet.

Author: We will say that they are planned, since we want to get the newsletter out.

> * We had our second Fortran Monthly call on July ?? (FIXME).

Reviewer 3: ```suggestion
* We had our second Fortran Monthly call on July 16.
```

> You can read watch the recording below (FIXME update link):
> 
> <iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/i-gRNGRzugc" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

Reviewer 3: ```suggestion
You can read watch the recording below:

<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/_ojLReFvjbs" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

---
layout: post
title: "Fortran newsletter: August 2020"
category: newsletter

> author: Ondřej Čertík, Milan Curcic, Laurence Kedward, and Jérémie Vandenplas

Author: ```suggestion
author: Ondřej Čertík, Milan Curcic, Laurence Kedward, Jérémie Vandenplas and Arjen Markus
```

  
Ongoing work:

* [#117](https://github.com/fortran-lang/fortran-lang.org/issues/117): Adding a
  Benchmarks section, a new dedicated repository was created at
  https://github.com/fortran-lang/benchmarks and many details have been
  discussed in [issues](https://github.com/fortran-lang/benchmarks/issues) there

[Let us know](https://github.com/fortran-lang/fortran-lang.org/issues) if you have any suggestions for the website and its content.
We welcome any new contributors to the website and the tutorials page in particular - see the [contributor guide](https://github.com/fortran-lang/fortran-lang.org/blob/master/CONTRIBUTING.md) for how to get started.

## Fortran Standard Library

What's new in the Fortran Standard Library:
[#223](https://github.com/fortran-lang/stdlib/pull/223): the structure of the Fortran Standard Library has been modified for clarity and ease of use. With these changes, both experimental and stable procedures will reside together in the same modules. The status of the procedures (experimental vs stable) are documented in the code, in the specs, and in the [website](https://stdlib.fortran-lang.org/)

## Fortran Package Manager

What's new in fpm:

> 

Author: ```suggestion
* [fpm-registry](https://github.com/fortran-lang/fpm-registry) created, please see the README there how to contribute a package
* [#146](https://github.com/fortran-lang/fpm/issues/146): we had a video discussion about the design of the new Fortran rewrite
```

Author: @everythingfunctional, anything else we should mention for fpm?

fpm is still in very early development, and we need as much help as we can get.
Here's how you can help today:

* Try to use it. Does it work? No? Let us know! Read the [fpm packaging guide](https://github.com/fortran-lang/fpm/blob/master/PACKAGING.md) to learn how to build your package with fpm.
* Browse the [open issues](https://github.com/fortran-lang/fpm/issues) and see if you can help implement any fixes or features.
* Adapt your Fortran package for fpm.
* Improve the documentation.

The short term goal of fpm is to make development and installation of Fortran packages with dependencies easier.
Its long term goal is to build a rich and decentralized ecosystem of Fortran packages and create a healthy
environment in which new open source Fortran projects are created and published with ease.

## Events

* [FortranCon 2020](https://tcevents.chem.uzh.ch/event/12) was held July 2 - 4.
with many interesting talks.
See the talk schedule
[here](https://tcevents.chem.uzh.ch/event/12/timetable/#20200702.detailed).
FIXME: is recordings available?

> * We had our second Fortran Monthly call on July 16.

Reviewer 2: ```suggestion
* We had our third Fortran Monthly call on July 16.
```
